### PR TITLE
Enable parallel execution of plugin generation

### PR DESCRIPTION
**Pull request by emcfarlane**
Run each plugin concurrently.

### CODE REVIEW

File: cmd/protoc-gen-multi/main.go [hunk 1]
```diff
@@ -69,26 +70,24 @@ func generate(ctx context.Context, args []string, request *pluginpb.CodeGenerato
...
-			return err
-		}
+		group.Go(func() error {
+			return plugin.generate(ctx, pluginRequest, pluginResponse)
+		})
+	}
+	if err := group.Wait(); err != nil {
+		return err
+	}
+	for i := range outs {
```
Comment (mfridman) [inline cmd/protoc-gen-multi/main.go:91]: ~I guess that's one way to do it, another option would be to range over a channel gathering the code gen response.~ Nah, that way you have it is probably cleaner, ignore me.

**Review by mfridman — APPROVED** [review]
lgtm, would be interesting to validate the delta for a larger module, like https://buf.build/envoyproxy/envoy

File: cmd/protoc-gen-multi/main.go [hunk 2]
```diff
@@ -131,19 +134,44 @@ func parsePlugins(args []string) (map[string]plugin, error) {
...
-			plugins[name] = plugin
 		} else {
 			return nil, fmt.Errorf("expected suffix \"_opt\" or \"_out\": %q", flag)
 		}
 	}
 	return plugins, nil
 }
+
+// generate executes the plugin with the given request and unmarshals the response.
+// Modifies both the request and response in place.
```
Comment (mfridman) [inline cmd/protoc-gen-multi/main.go:156]: We may want to surface the name of the failing plugin, maybe a deferred `retErr` check that includes the name of the plugin, e.g., `fmt.Errorf("failed to execute plugin: %s: %w, protoc-gen-name, err)`

 That way the caller has a chance to debug the specific plugin.

Comment (emcfarlane) [reply cmd/protoc-gen-multi/main.go:156]: Added it at the call site.

**Comment by emcfarlane** [comment]
Played a little locally with generating envoy with the plugins go, grpc, and vt protos. Not much in it! The time to execute a single CodeGeneratorRequest is usually small ~10ms, overall time of 5s. Will depend on plugins being invoked.

**Review by mfridman — APPROVED** [review]

**Pull request merged**